### PR TITLE
Rewrite to Js

### PR DESCRIPTION
**Pull request by havardh**

### CODE REVIEW

File: parser/config.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import type { WorkspaceConfig, NodeConfig } from '../index';
```
Comment (relekang) [inline parser/config.js]: Add `@flow` ?

File: parser/args.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        args,
+      );
+    }
+  } else {
+    throw new InvalidConfig(
+      `expected config.args to be of type 'string' or 'array, got: ${typeof config}'`, config);
+  }
+}
+
+export default function parseArgs(
```
Comment (relekang) [inline parser/args.js]: Why not use a arg parser lib?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    "eslint-config-airbnb-base": "^11.1.3",
+    "eslint-plugin-import": "^2.2.0",
+    "flow-bin": "^0.44.2"
+  },
+  "dependencies": {
+    "i3": "^0.2.1",
+    "shelljs": "^0.7.7",
+    "tmp": "^0.0.31",
+    "chai": "^3.5.0",
+    "mocha": "^3.3.0",
```
Comment (relekang) [inline package.json]: Dev dependency?

File: wms/i3.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
+import { createClient } from 'i3';
```
Comment (relekang) [inline wms/i3.js]: Add `@flow`?

File: wms/i3.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+import { fileSync } from 'tmp';
+import { writeFileSync } from 'fs';
```
Comment (relekang) [inline wms/i3.js]: We should avoid using sync.

File: .gitignore [hunk 6]
```diff
@@ -1,98 +1 @@
...
-
-# Spyder project settings
-.spyderproject
-
-# Rope project settings
-.ropeproject
-
-# mkdocs documentation
-/site
+node_modules
```
Comment (relekang) [inline .gitignore]:
```shell
curl https://raw.githubusercontent.com/github/gitignore/master/Node.gitignore > .gitignore 
```
❓

File: parser/args.js (hunk 2 above)
Comment (havardh) [inline parser/args.js]: I've created #15 for this.

File: wms/i3.js (hunk 4 above)
Comment (havardh) [inline wms/i3.js]: This proved a bit hard to do. I think we need to rethink/restructure the flow types a bit.

**Pull request merged**